docs/blog/2024-12-10-tailoring-nx-for-your-organization.md

```diff
@@ -0,0 +1,182 @@
+
+```bash
+nx generate @nx/react:library --name shared-util --directory libs/shared/util --importPath="@org/shared/util" --tags=type:util,scope:shared
+```
+
+To understand the right options for this generator, the engineer needs to remember:
+
+- the project naming convention based on the type of library and its scope
+- the directory structure based on nested folders for scope
+- the import naming convention
+- tag name convention based on type and scope
+
+This is a lot to remember! Even the most diligent engineer will forget one of these or miss a typo. **A custom generator** can solve this problem. Consider a custom generator for your organization that accepts options like this:
+
+```bash
+nx generate @org/plugin:library --name util --scope shared --type util
+```
+
+This custom generator takes well-defined options for your org and translates them into the options needed for the underlying generator. A snippet of that would look something like this:
+
+```typescript
+import { Tree } from '@nx/devkit';
+import { Linter } from '@nx/eslint';
+import { libraryGenerator as reactLibraryGenerator } from '@nx/react';
+import { LibraryGeneratorSchema } from './schema';
+
+export async function libraryGenerator(
+  tree: Tree,
+  options: LibraryGeneratorSchema // schema with strong types for scope and type
+) {
+  // create project name based on submitted name and type
+  const projectName = `${options.type}-${options.name}`;
+  //determine directory based on scope and project name
+  const directory = joinPathFragments('libs', options.scope, projectName);
+  // determine import path based on scope and project name
+  const importPath = `@org/${scope}/${projectName}`;
+  // run the @nx/react library generator with options set
+  const callbackAfterFilesUpdated = await reactLibraryGenerator(tree, {
+    name: projectName,
+    directory,
+    importPath,
+    tags: [`type:${options.type}`, `scope:${shared}`],
+    linter: Linter.EsLint,
+    style: 'css',
+    unitTestRunner: 'vitest',
+  });
+
+  return callbackAfterFilesUpdated;
+}
+
+export default libraryGenerator;
+```
+
+This example of a generator doesn’t do much but pass in options to an existing generator. However, it will consistently:
+
+- Name the project appropriately
+- Create the correct import path
+- Place new projects in the correct directory
+- Add the proper tags to the project
+- Select the correct set of tools
+
+All of that without having to refer to your documentation on how to run the generator the right way. And most of the functionality is provided by Nx! You just need to provide some logic specific to your organization.
+
+These generators will also show up in [Nx Console](/getting-started/editor-setup) alongside the other generators from Nx plugins, so engineers will have quick access.
+
+{% youtube src="https://www.youtube.com/watch?v=myqfGDWC2go" /%}
+
+## Configure your tools in one place with custom inferred tasks
```

Comment:
I think we need to do some more explanation here. Inferred tasks sounds like super complicated. I'd mention things like
- project crystal
- inference plugins can be configured once at the root level and then automatically whenever that specific "config" file it monitors is found, you can perform some action (whatever that might be)

> Writing your own task inference plugin allows you handle these situations. Let’s say the hottest, coolest new tool hits the ecosystem, and you need to use it today.

I'd not mention it in this way. Even today if a new tool hits the market and you add it to an Nx workspace you can run it right away by defining a package.json scripts and go. But then you can create a Crystal plugin to using task inference to automate how the new tool is used within your monorepo, s.t. not every single new project needs to redefine it etc